### PR TITLE
MCKIN-13610 Add logs to LTI launch erros

### PR DESCRIPTION
**Pull request by moeez96**
Log any errors in the LTI launch flow to splunk.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @moeez96! I've created [OSPR-4264](https://openedx.atlassian.net/browse/OSPR-4264) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: lms/djangoapps/lti_provider/views.py [hunk 1]
```diff
@@ -48,12 +48,22 @@ def lti_launch(request, course_id, usage_id):
           pair
     """
     if not settings.FEATURES['ENABLE_LTI_PROVIDER']:
+        log.error(
+            'LTI Provider not enabled in settings for course %s and usage_id %s',
+            course_id,
+            usage_id,
```
Comment (wajeeha-khalid) [inline lms/djangoapps/lti_provider/views.py]: What is usage ID here? is it client ID for LTI config? If yes, can we skip its logging?

File: lms/djangoapps/lti_provider/views.py [hunk 2]
```diff
@@ -48,12 +48,22 @@ def lti_launch(request, course_id, usage_id):
...
         return HttpResponseForbidden()
 
     # Check the LTI parameters, and return 400 if any required parameters are
     # missing
     params = get_required_parameters(request.POST)
     if not params:
+        log.error(
+            'LTI required parameters missing for course %s and usage_id %s.',
+            course_id,
+            usage_id,
```
Comment (wajeeha-khalid) [inline lms/djangoapps/lti_provider/views.py]: Dito

File: lms/djangoapps/lti_provider/views.py [hunk 3]
```diff
@@ -65,10 +75,22 @@ def lti_launch(request, course_id, usage_id):
             params['oauth_consumer_key']
         )
     except LtiConsumer.DoesNotExist:
+        log.error(
+            'LTI Consumer not found for course %s, usage_id %s and consumer key %s.',
+            course_id,
+            usage_id,
+            params['oauth_consumer_key'],
+        )
```
Comment (wajeeha-khalid) [inline lms/djangoapps/lti_provider/views.py]: Same as above but for both `usage_id` and `oauth_consumer_key`

File: lms/djangoapps/lti_provider/views.py [hunk 4]
```diff
@@ -65,10 +75,22 @@ def lti_launch(request, course_id, usage_id):
...
         return HttpResponseForbidden()
 
     # Check the OAuth signature on the message
     if not SignatureValidator(lti_consumer).verify(request):
+        log.error(
+            'LTI Consumer not valid for course %s, usage_id %s and consumer key %s.',
+            course_id,
+            usage_id,
+            params['oauth_consumer_key'],
```
Comment (wajeeha-khalid) [inline lms/djangoapps/lti_provider/views.py]: dito

**Comment by moeez96** [comment]
@wajeeha-khalid The extra info is logged for ease of tracking.
In case of a failure, we wont be dependent on time of failure to track the error in splunk logs. We can apply filter on course id, usage id and in cases where applicable, the consumer key.
The main idea of error logging is to trace cause of failure effectively, hence the info.

**Pull request merged**